ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/AssertStatement.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.parser.tree;
+
+import io.confluent.ksql.parser.Node;
+import io.confluent.ksql.parser.NodeLocation;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Indicates that the given {@link io.confluent.ksql.parser.Node} is used
+ * for testing purposes.
+ */
+public abstract class AssertStatement<T extends Statement> extends Node  {
+
+  private final T statement;
+
+  protected AssertStatement(
+      final Optional<NodeLocation> location,
+      final T statement
+  ) {
+    super(location);
+    this.statement = Objects.requireNonNull(statement, "statement");
+  }
+
+  public T getStatement() {
+    return statement;
+  }
+}
```

Comment:
it seems strange to me that `AssertStatement` is template on a `Statement` type.  There doesn't currently seem to be any logic that calls `getStatement` at present, (though I'm sure you've got some planned).  I don't see why this is necessary. Can not each assert standalone and know what it needs to check? What does the generic template arg give us?

Why not just:

```suggestion
public abstract class AssertStatement extends Node  {

  protected AssertStatement(
      final Optional<NodeLocation> location
  ) {
    super(location);
  }
}
```

Reply:
makes sense, again given the stacked PR I'll address this in a follow-up (pinky promise!)